### DISCUSSION TITLE
SUP: can not fly,     ESTKALMAN: State out of bounds, resetting

### DISCUSSION THREAD

**Discussion by surendhar091992** [bitcraze/discussions#1078] [Q&A]
I am using 4 cf2 in my experiment (simple cmdvel based waypoints), and i am using optitrack(flex-13) motion capture system. I am using crazyswarm package (ros-noetic based). The moment i get the 'ESTKALMAN: State out of bounds, resetting' error, the cf2 crashes. i would like to know the reason for this particular error.

I have attached the terminal error snapshot and (roll,pitch,yaw) from tf2 command.

Terminal error plot:
![err_plot](https://github.com/bitcraze/discussions/assets/104561072/fb7f1ac2-b649-45c9-a631-9a2aab05beb6)

roll,pitch,raw plot from tf2 command:

![Screenshot from 2023-12-08 14-19-19](https://github.com/bitcraze/discussions/assets/104561072/144110f0-b661-4e67-9512-6b52fd00f8ad)

Whenever the roll,pitch,yaw stays close to zero (even from the takeoff), the drone crashes in the middle of the flight.

**Comment by knmcguire** [comment]
Seems like something is going wrong with the optitrack data. You are using Crazyswarm2 I see, have you tried asking this question there? They maintain both the crazyswarm2 package as the lib motion capture that you are using: https://github.com/IMRCLab/crazyswarm2/discussions

From our side the only thing we can advise is to try it with one with our mocap example in our python library: https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/mocap/mocap_hl_commander.py

**Reply by surendhar091992** [reply]
Thank you for getting back, I will look into that example. Also, any examples(mobcap/non-mocap) for sending velocity commands ?

**Reply by krichardsson** [reply]
Just to fill in some info, you get the `ESTKALMAN: State out of bounds, resetting` message when the state in the kalman estimator has "unreasonable" values. The default settings are position outside +-100m or velocity higher than 10 m/s
